mozetl/taar/taar_dynamo.py

```diff
@@ -0,0 +1,418 @@
+
+    jsonDataRDD = clientsData.select("city",
+                                     "subsession_start_date",
+                                     "subsession_length",
+                                     "locale",
+                                     "os",
+                                     "places_bookmarks_count",
+                                     "scalar_parent_browser_engagement_tab_open_event_count",
+                                     "scalar_parent_browser_engagement_total_uri_count",
+                                     "scalar_parent_browser_engagement_unique_domains_count",
+                                     "active_addons",
+                                     "disabled_addons_ids").toJSON()
+
+    print("jsonDataRDD selected")
+    rdd = subset.rdd.zip(jsonDataRDD)
+    print("subset rdd has been zipped")
+
+    # Filter out any records with invalid dates or client_id
+    filtered_rdd = rdd.filter(filterDateAndClientID)
+    print("rdd filtered by date and client_id")
+
+    # Transform the JSON elements into a 4-tuple as per docstring
+    merged_filtered_rdd = filtered_rdd.map(list_transformer)
+    print("rdd has been transformed into tuples")
+
+    return merged_filtered_rdd
+
+
+def load_rdd(prod_iam_role, region_name, table_name, rdd):
+    # Apply a MapReduce operation to the RDD
+    dynReducer = DynamoReducer(prod_iam_role, region_name, table_name)
+
+    reduction_output = rdd.reduce(dynReducer.dynamo_reducer)
+    print("1st pass dynamo reduction completed")
+
+    # Apply the reducer one more time to force any lingering
+    # data to get pushed into DynamoDB.
+    final_reduction_output = dynReducer.dynamo_reducer(reduction_output,
+                                                       EMPTY_TUPLE,
+                                                       force_write=True)
+    return final_reduction_output
+
+
+def run_etljob(spark, run_date, region_name, table_name, prod_iam_role, sample_rate):
+    reduction_output = etl(spark,
+                           run_date,
+                           region_name,
+                           table_name,
+                           prod_iam_role,
+                           sample_rate)
+    report_data = (reduction_output[0], reduction_output[1])
+    print("=" * 40)
+    print("%d records inserted to DynamoDB.\n%d records remaining in queue." % report_data)
+    print("=" * 40)
+    return reduction_output
+
+
+@click.command()
+@click.option('--date', required=True)  # YYYYMMDD
+@click.option('--region', default='us-west-2')
+@click.option('--table', default='taar_addon_data_20180206')
+@click.option('--prod-iam-role', default='arn:aws:iam::361527076523:role/taar-write-dynamodb-from-dev')
+@click.option('--sample-rate', default=0)
+def main(date, region, table, prod_iam_role, sample_rate):
+    APP_NAME = "HBaseAddonRecommenderView"
+    conf = SparkConf().setAppName(APP_NAME)
+    conf = conf.setMaster("local[*]")
```

Comment:
If you set the master to local on a spark cluster this will only run using the master node -- our EMR clusters already have `spark.master` set correctly to the YARN cluster manager so you don't need to set this conf

Reply:
Cool.  Thanks, I've removed this now.